packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -408,15 +409,74 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
       },
       select: {
         userId: true,
+        eventType: {
+          select: {
+            teamId: true,
+          },
+        },
+        attendees: {
+          select: {
+            email: true,
+          },
+        },
       },
     });
-    hosts = hosts.filter((host) => host.user.id === originalRescheduledBooking?.userId || 0);
+    if (
+      eventType.rescheduleWithSameRoundRobinHost &&
+      eventType.schedulingType === SchedulingType.ROUND_ROBIN
+    ) {
+      hosts = hosts.filter((host) => host.user.id === originalRescheduledBooking?.userId || 0);
+    }
+    const userReschedulingIsOwner = isUserReschedulingOwner(
+      user?.id,
+      originalRescheduledBooking?.userId || 0
+    );
+    let isDynamicEventAndUserIsOwner = false;
+    if (input.usernameList && input.usernameList.length > 1 && user) {
+      isDynamicEventAndUserIsOwner = eventType.users.some((eventUser) => eventUser.id === user.id);
+    }
+    let isTeamOwnerOrAdmin = false;
+    if (input?.isTeamEvent && originalRescheduledBooking?.eventType?.teamId) {
+      const teamOwnerOrAdmin = await prisma.membership.findFirst({
+        where: {
+          teamId: originalRescheduledBooking?.eventType?.teamId,
+          userId: user?.id,
```

Comment:
In case bookingOrganizer is in an organization and user rescheduling is org owner/admin, they should also be able to reschedule the booking.

This currently only works in case of team admin/owner.

Reply:
Done. As per my understanding, org owner or admin can only reschedule team events so I am only checking for org owner or admin for team events of the org. Please let me know if that is not the case.